test-runner/src/main/scala/scala/scalanative/testinterface/ProcessRunner.scala

```diff
@@ -0,0 +1,47 @@
+package scala.scalanative.testinterface
+
+import java.io.File
+import scala.concurrent.{Future, Promise}
+import scala.scalanative.build.Logger
+
+private[testinterface] class ProcessRunner(executableFile: File,
+                                           envVars: Map[String, String],
+                                           args: Seq[String],
+                                           logger: Logger,
+                                           port: Int)
+    extends AutoCloseable {
+
+  private[this] val process = {
+    val builder =
+      new ProcessBuilder(executableFile.toString +: port.toString +: args: _*)
+        .inheritIO()
+
+    envVars.foreach {
+      case (k, v) =>
+        builder.environment().put(k, v)
+    }
+
+    builder.start()
+  }
+
+  private[this] val runnerPromise: Promise[Unit] = Promise[Unit]()
+  private[this] val runner = new Thread {
+    setName("TestRunner")
+    override def run(): Unit = {
+      logger.info(s"Starting process '$executableFile' on port '$port'.")
```

Comment:
Consider moving this log line to the "main" thread, before `builder.start()`, as it seems to be more applicable there.